codeworld-compiler/exec/Main.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PackageImports    #-}
+
+{-
+  Copyright 2017 The CodeWorld Authors. All rights reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-}
+
+import Options.Applicative
+import Control.Monad (join)
+import Data.Monoid ((<>))
+import System.Environment
+import System.Directory
+import System.IO
+
+import "codeworld-compiler" Compile as C
```

Comment:
I'm still confused about the use of package imports.  What other package has a Compile module?\

Also, you should always write one of:

    import qualified Compile as C

or:

    import Compile

Adding `as C` without `qualified` imports symbols BOTH as `C.foo` and `foo`, which is almost never what you wanted.

In this case, I doubt a qualified import is needed, so I'm tempted to recommend the latter.